### ISSUE TITLE
Not all compctl completions working

### ISSUE THREAD

**Issue by marcbachmann** [warpdotdev/Warp#2179]
### Describe the bug

`compctl` definitions don't work together with the suggestions when tabbing.

### To Reproduce

```bash
mkdir -p $HOME/Development/foo $HOME/Development/bar
compctl -W $HOME/Development/ -/ dev

# Type 'dev' & tab twice
# It doesn't suggest the two directories
```

### Expected behaviour

```bash
# type
dev [tab] [tab]

# should suggest both directories
foo
bar
```

### Screenshots

In iTerm it suggests both directories
<img width="136" alt="image" src="https://user-images.githubusercontent.com/431376/202730010-c266c7a7-5503-4c3e-8183-1e5a7f7b7b41.png">

In warp it ignores the definition
<img width="766" alt="image" src="https://user-images.githubusercontent.com/431376/202732092-0fe96703-3d49-45e9-9d81-4a10aff40f57.png">

### Operating System

MacOS

### OS Version

13.0

### Shell Version

zsh 5.8.1 (x86_64-apple-darwin22.0)

### Warp Version

v0.2022.11.14.14.55.stable_01

### Does this block you from using Warp daily?

Yes

**Comment by nichoj10** [comment]
Hi! Thanks for filling this! Warp should pick up this completion. If it recognizes `dev` in a file context, tab should autocomplete to the full file path. For example, from ~, `cd dev[tab]` --> `cd Development/`. Do you mind explaining your use case for `dev [tab] [tab]` so we can understand the desired behavior better?

**Comment by marcbachmann** [comment]
> Do you mind explaining your use case for dev [tab] [tab] so we can understand the desired behavior better?

I'm working on many github projects.
Mapping the shortcuts to the specific directories (which use the github org as name).
```
~/Development/some-github-org
~/Development/some-github-org/some-api
~/Development/some-github-org/some-other-api
```

in my case i'm not using `cd` in front of the shortcuts. just execute `li[tab][tab]`.
```bash
gotorepo () {
	local dir=$HOME/Development/$1
	[ -d "$dir/$2/master" ] && cd $dir/$2/master && return
	[ -d "$dir/$2" ] && cd $dir/$2 && return
	echo "Failed to go to $dir/$2"
}

function li () { gotorepo livingdocsIO $1; }
compctl -W $HOME/Development/livingdocsIO/ -/ li
```

So you're mentioning file context, doesn't `compctl -W $HOME/Development/livingdocsIO/ -/ li` define that in my case?

**Comment by lucianoratamero** [comment]
I'm having the same issue with custom functions + completions, commenting here to get updates :]
(thanks for the hard work!)

**Comment by nichoj10** [comment]
Gotcha! In that case, my above answer is misleading. Sorry for that!

Warp bypasses shell-native completions altogether so we don't provide support for zsh-generated completions using either `compctl` or completion widgets. We've had a good bit of discussion around this and have it in our roadmap to allow for custom completions that are built into Warp's own completion engine. We don't yet have an ETA for when this will be built. 

A little more information: The reason for this is because of Warp's IDE-like text editing. We don't round-trip with the shell (zsh) on a `tab` press and then attempt to parse and re-render based on the the shell's output - mostly because of performance issues. Instead, the autosuggestion and completion behavior operates at the terminal level without shell reliance.

**Comment by prime-optimal** [comment]
I really hope this gets implemented because I think Warp with oh-my-zsh autocomplete would be an unbeatable combination

**Comment by Saafo** [comment]
> I really hope this gets implemented because I think Warp with oh-my-zsh autocomplete would be an unbeatable combination

Agree, currently there's no way for me to use completions generating by the process. It's important to me and had already become a shortcomings of Warp completions. If you don't want to make it compatible with `compctl` or zsh completion, at least give some way ASAP to solve the basic demand. Half a year passed!

**Comment by An0nymous0** [comment]
+1
